module/VuFind/src/VuFindTest/Integration/MinkTestCase.php

```diff
@@ -294,17 +294,39 @@ protected function getVuFindUrl($path = '')
     /**
      * Get query string for the current page
      *
+     * @param bool $excludeSid Whether to remove any sid from the query string
+     *
      * @return string
      */
-    protected function getCurrentQueryString(): string
+    protected function getCurrentQueryString(bool $excludeSid = false): string
     {
         return str_replace(
             ['%5B', '%5D', '%7C'],
             ['[', ']', '|'],
-            parse_url($this->getMinkSession()->getCurrentUrl(), PHP_URL_QUERY)
+            parse_url(
+                $excludeSid ? $this->getCurrentUrlWithoutSid()
+                    : $this->getMinkSession()->getCurrentUrl(),
+                PHP_URL_QUERY
+            )
         );
     }
 
+    /**
+     * Get current URL without any sid parameter in the query string
+     *
+     * @return string
+     */
+    protected function getCurrentUrlWithoutSid(): string
+    {
+        $this->getMinkSession();
+        $url = $this->getMinkSession()->getCurrentUrl();
+        $url = preg_replace('/\?sid=\d+/', '?', $url);
```

Comment:
If there's a question mark, we need to keep it, but if we also keep the ampersand, it can lead to a dangling one:
`path?sid=1&foo=2` => `path?&foo=2`
or
`path?foo=2&sid=1` => `path?foo=2&`

While these technically work fine, they don't match the expected results. I trimmed the other preg_replace from the method, but had to keep the rest.

Reply:
Good point. How about this instead?

```
$url = preg_replace('/([&?])sid=[^&]*&?/', '$1', $url);
```

That should take care of ampersand trimming in a one-liner, plus it will also remove empty parameters or parameters with unexpected values in them, so I think it's a little more universal than assuming that the value will always be numeric. We could even generalize this into a "remove query parameter" method that takes a parameter name, if we eventually need this for other purposes.

Reply:
Yes, that'd work. Now applied!